File: app/src/main/java/com/fractalwrench/crazycats/network/responses/SubredditListingResponse.java
package com.fractalwrench.crazycats.network.responses;

import java.util.List;

public class SubredditListingResponse {

    private ListingData data;

    public static class ListingData {
        private List<ListingChildren> children;

        public List<ListingChildren> getChildren() {
            return children;
        }

        public void setChildren(List<ListingChildren> children) {
            this.children = children;
        }
    }

    public static class ListingChildren {
        private SubredditThreadResponse data;

        public SubredditThreadResponse getData() {
            return data;
        }

        public void setData(SubredditThreadResponse data) {
            this.data = data;
        }
    }

    public ListingData getData() {
        return data;
    }

    public void setData(ListingData data) {
        this.data = data;
    }
}

File: app/src/main/java/com/fractalwrench/crazycats/network/responses/SubredditListingResponse.kt
package com.fractalwrench.crazycats.network.responses

data class SubredditListingResponse(val data: ListingData? = null)

data class ListingData(val children: List<ListingChildren>?)

data class ListingChildren(val data: SubredditThreadResponse?)

data class SubredditThreadResponse(val id: String?,
                                   val url: String?,
                                   val title: String?,
                                   val preview: Preview?)

